src/strongbox/ui/cli.clj

```diff
@@ -291,22 +292,37 @@
   [updateable-addon-list :addon/installable-list]
   (run! core/install-addon-guard-affective updateable-addon-list))
 
+(defn-spec addon-locks set?
+  [addon :addon/installable, downloaded-file ::sp/extant-archive-file]
+  (let [;; addon may not be installed yet, we may not have any `:dirname` values at all
+        existing-dirs (->> addon addon/flatten-addon (map :dirname) utils/nilable set)
+        strip-trailing-slash #(utils/rtrim % "/")
+        new-dirs (->> downloaded-file
+                      zip/zipfile-normal-entries
+                      zip/top-level-directories
+                      (map :path)
+                      (map strip-trailing-slash)
+                      set)]
+    (into existing-dirs new-dirs)))
+
 (defn-spec install-update-these-in-parallel nil?
   "installs/updates a list of addons in parallel, pushing guard checks into threads and then installing serially."
   [updateable-addon-list :addon/installable-list]
   (let [queue-atm (core/get-state :job-queue)
         install-dir (core/selected-addon-dir)
+        current-locks (atom #{})
         add-download-job! (fn [addon]
                             (let [job-fn (fn []
-                                           [addon (core/download-addon-guard-affective addon install-dir)])
+                                           (let [downloaded-file (core/download-addon-guard-affective addon install-dir)
+                                                 locks-needed (addon-locks addon downloaded-file)]
+                                             (utils/with-lock current-locks locks-needed
+                                               (core/install-addon-affective addon install-dir downloaded-file))
+                                             (core/refresh-addon addon)))
```

Comment:
move this under the with-lock

it's possible for two sets of changes to be forced to happen sequentially because of the locks and for the refresh to read dirty data